src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1677,13 +1682,22 @@ void uct_dc_mlx5_iface_set_ep_failed(uct_dc_mlx5_iface_t *iface,
         return;
     }
 
-    ep->flags |= UCT_DC_MLX5_EP_FLAG_ERR_HANDLER_INVOKED;
+    if (ep->flags & UCT_DC_MLX5_EP_FLAG_ERR_HANDLER_INVOKED) {
+        return;
+    }
```

Comment:
can we move this back before checking fc ep?

Reply:
since we don't need to do flush(CANCEL) or invoke error handler for FC EP, I guess it worth checking FC EP first

Reply:
FC endpoint can't enter the body of this `if` statement.
need to check `UCT_DC_MLX5_IFACE_FLAG_FC_EP_FAILED` before printing a debug message - done